### ISSUE TITLE
Errors due to malformed history files should be caught

### ISSUE THREAD

**Issue by Danmou** [xonsh/xonsh#2518]
I somehow got a corrupted history file, which caused a traceback to be printed a few seconds after starting xonsh, and again at random intervals.

```
Traceback (most recent call last):
  File "/usr/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.5/dist-packages/xonsh/ptk/history.py", line 54, in run
    for cmd in hist.all_items():
  File "/usr/local/lib/python3.5/dist-packages/xonsh/history/__amalgam__.py", line 618, in all_items
    commands = json_file.load()['cmds']
  File "/usr/local/lib/python3.5/dist-packages/xonsh/__amalgam__.py", line 752, in load
    return self._load_or_node(offset, size)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/__amalgam__.py", line 759, in _load_or_node
    val = json.loads(s)
  File "/usr/lib/python3.5/json/__init__.py", line 319, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.5/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.5/json/decoder.py", line 355, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Unterminated string starting at: line 2 column 1493 (char 1504)
```

Removing the file solved the problem, but tracking down the problematic file was rather difficult. I'm not sure if xonsh should just completely ignore this kind of error,  but if not it should at least give a brief, understandable error, and include the name of the corrupted file.

**Comment by scopatz** [comment]
Thanks for reporting @Danmou! This seems like a bug, for sure.

**Comment by scopatz** [comment]
Potential fix in #2527

**Comment by gforsyth** [comment]
Hey @Danmou -- the fix for this is now merged in.  Please re-open or start a new issue if the problem persists.  Thanks!
